pages/register.tsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react'
+import Head from 'next/head'
+import RegisterForm from '../components/RegisterForm'
+import styles from '../styles/Login.module.css'
+import router from 'next/router'
+
+const Register = () => { 
+  const registerRequest = (formData: React.FormEvent<HTMLInputElement>) => {
+    fetch("http://localhost:3001/api/signup", {

```

Comment:
I recommend creating a file called `api/base`. This can hold all the commonly used functions for each API request. Namely:
- baseUrl
- authHeaders

```
const baseUrl = () => process.env.NEXT_PUBLIC_API

const authHeaders = () => {
  // logic to get auth key
  return { Authorization: 'get authorization key' }
}
```

With that, you can make use of these by:

```

const register = (formdata) => {
  return fetch(`${baseUrl}/api/register`, method: "post", headers: { "Content-Type": "application/json", ...authHeaders() }
}

```